File: myfragments/src/main/java/pl/mareklangiewicz/myfragments/MyExampleFragment.java
package pl.mareklangiewicz.myfragments;

import android.app.Fragment;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import pl.mareklangiewicz.myviews.MyExampleView;

import static pl.mareklangiewicz.myutils.MyTextUtils.str;


public final class MyExampleFragment extends Fragment {
    private static final String ARG_TEXT = "text";

    public MyExampleFragment() {}

    public static MyExampleFragment newInstance(String text) {
        MyExampleFragment fragment = new MyExampleFragment();
        Bundle args = new Bundle();
        args.putString(ARG_TEXT, text);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.my_example_fragment, container, false);
        MyExampleView mev = (MyExampleView) root.findViewById(R.id.my_example_view);
        Bundle args = getArguments();
        if(args != null) {
            StringBuilder text = new StringBuilder();
            for(String key: args.keySet()) {
                text.append(key);
                text.append(": ");
                text.append(str(args.get(key)));
                text.append("\n");
            }
            mev.setText(text.toString());
        }
        return root;
    }
}

File: myfragments/src/main/java/pl/mareklangiewicz/myfragments/MyExampleFragment.kt
package pl.mareklangiewicz.myfragments

import android.app.Fragment
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.mf_my_example_fragment.*
import pl.mareklangiewicz.myutils.str


open class MyExampleFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        super.onCreateView(inflater, container, savedInstanceState) //just for logging
        return inflater.inflate(R.layout.mf_my_example_fragment, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        arguments?.let {
            val text = StringBuilder()
            for (key in it.keySet()) text.append("$key: ${it.get(key).str}\n")
            my_example_view.text = text.toString()
        }
    }
}
